### PR TITLE
qutebrowser: update to 2.0.1

### PR DESCRIPTION
**Pull request by Byl3x**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by ericonr** [comment]
Please add `checkdepends="$depends"` to the template.

**Comment by Byl3x** [comment]
I don't understand, my native arch is x86_64 and building it in a vm(that didn't have the previous version or anything related) with xbps-src works too

**Comment by ericonr** [comment]
This is running the tests, which you can do by running `./xbps-src -Q pkg qutebrowser`.

Also, please squash the commits.

File: srcpkgs/qutebrowser/template [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
 pkgname=qutebrowser
-version=1.14.1
+version=2.0.0
 revision=1
 build_style=python3-module
 hostmakedepends="python3-setuptools asciidoc"
 depends="python3-PyQt5-quick python3-Jinja2 python3-Pygments python3-pyPEG2
  python3-yaml python3-attrs python3-PyQt5-opengl python3-PyQt5-sql
  qt5-plugin-sqlite python3-setuptools"
+ checkdepends="$depends"
```
Comment (ericonr) [inline srcpkgs/qutebrowser/template:10]: try adding `python3-pytest`

**Comment by Byl3x** [comment]
Well the tests are causing the issue and when I look at the build logs for the older versions they skipped the tests, so I'm wondering if the previous versions passed the tests, I'll try it

**Comment by FollieHiyuki** [comment]
Should we do something to #27490 too? qutebrowser 2.0.0 comes with the adblocking feature

**Comment by pbui** [comment]
2.0.1 has been released: https://github.com/qutebrowser/qutebrowser/releases/tag/v2.0.1

**Comment by Byl3x** [comment]
So I tried running the test and found out that the test needs a running Xorg server to function(it displays a window) and it is also deprecated, I'm not sure how to change the build so that it doesn't run the tests

**Comment by ericonr** [comment]
Please apply this patch:

```diff
diff --git a/srcpkgs/qutebrowser/template b/srcpkgs/qutebrowser/template
index c80d32792f..4c160a1adf 100644
--- a/srcpkgs/qutebrowser/template
+++ b/srcpkgs/qutebrowser/template
@@ -7,7 +7,7 @@ hostmakedepends="python3-setuptools asciidoc"
 depends="python3-PyQt5-quick python3-Jinja2 python3-Pygments python3-pyPEG2
  python3-yaml python3-attrs python3-PyQt5-opengl python3-PyQt5-sql
  qt5-plugin-sqlite python3-setuptools"
- checkdepends="$depends python3-pytest"
+checkdepends="$depends python3-pytest xvfb-run"
 short_desc="Keyboard-focused browser with a minimal GUI"
 maintainer="Érico Nogueira <ericonr@disroot.org>"
 license="GPL-3.0-or-later"
@@ -37,6 +37,12 @@ pre_build() {
 	a2x -f manpage doc/qutebrowser.1.asciidoc
 }
 
+do_check() {
+	# testing with pytest would be nice, but requires unpackaged plugins:
+	# pytest-bdd, pytest-benchmark, pytest-instafail, pytest-mock, pytest-qt, pytest-rerunfailures
+	:
+}
+
 post_install() {
 	vman doc/qutebrowser.1
 	vinstall misc/org.qutebrowser.qutebrowser.desktop 644 usr/share/applications
```

**Review by ahesford — CHANGES_REQUESTED** [review]
In addition to bumping to the fresh 2.0.1, you need to go through the [Quick checklist for packages](https://github.com/qutebrowser/qutebrowser/releases/tag/v2.0.0) provided by the project. The dependencies need to be brought in line with upstream recommendation.

Regarding `python3-Pygments` becoming optional, it would be nice to know where in qutebrowser it is used so we can decide if it should be retained as a dependency of this package.

**Comment by ahesford** [comment]
> +do_check() {
> +	# testing with pytest would be nice, but requires unpackaged plugins:
> +	# pytest-bdd, pytest-benchmark, pytest-instafail, pytest-mock, pytest-qt, pytest-rerunfailures
> +	:
> +}

@ericonr We do provide `python3-pytest-mock`. As for the others, are the tests that depend on them so numerous that the entire check should be skipped, or would it be feasible to `--ignore` or `--ignore-glob` the affected tests?

**Comment by ericonr** [comment]
@ahesford we also provide pytest-qt, fwiw.

The issue is that the tests seem to check for these plugins at startup, and it complains immediatelly about them not being there, instead of being runtime failures.

**Comment by Byl3x** [comment]
@ahesford python3-Pygments is used for syntax highlighting when viewing page source(only with :view-source --pygments or on WebKit) and also when formatting JSON files

**Comment by ahesford** [comment]
Syntax highlighting of source and JSON seems worth keeping the Pygments dependency. The other obsolete dependencies still need to be dropped.

**Comment by Byl3x** [comment]
I checked the dependencies, only python3-pyPEG2 shouldn't be required. Also the python3-Pygments are only used if you choose to use :view-source with the --pygments flag or when using a userscript that is in the official repo or when you use QtWebKit instead of QtWebEngine

**Comment by ahesford** [comment]
From the list:

> Remove the attrs (attr) dependency.

> Move the setuptools dependency from runtime (for pkg_resources) to build-time.

If the Pygments support really requires a `--pygments` flag (it doesn't in the current version---`:view-source` is definitely colored for me), then I would drop its requirement here, since having to explicitly opt in would mean that syntax coloring isn't the expected default behavior.

**Comment by Byl3x** [comment]
I didn't notice the attrs dependency removal, sorry. What I meant with the Pygments also was that to use it you have to do `:view-source --pygments` so I think that it's safe to drop the requirement

**Comment by paper42** [comment]
python3-pyPEG2 isn't used anywhere else, so it should be safe to drop the package (upstream dead)

File: srcpkgs/qutebrowser/template [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+version=2.0.1
 revision=1
 build_style=python3-module
 hostmakedepends="python3-setuptools asciidoc"
-depends="python3-PyQt5-quick python3-Jinja2 python3-Pygments python3-pyPEG2
- python3-yaml python3-attrs python3-PyQt5-opengl python3-PyQt5-sql
- qt5-plugin-sqlite python3-setuptools"
+depends="python3-PyQt5-quick python3-Jinja2 python3-yaml python3-PyQt5-opengl python3-PyQt5-sql
+ qt5-plugin-sqlite"
+checkdepends="$depends python3-pytest xvfb-run"
```
Comment (ahesford) [inline srcpkgs/qutebrowser/template:9]: If you're skipping tests, don't add checkdepends.

**Review by ahesford — APPROVED** [review]
This update is working for me.

**Comment by paper42** [comment]
What do you think about adding pdf.js as a dependency. It's an optional dependency which allows showing PDF files inside the browser (Ctrl+p in the download dialog).

**Comment by ahesford** [comment]
I wouldn't add that dependency. When the browser presents the option to display PDFs, I think it is pretty clear that it wants pdf.js, so users should know where to look if they want to enable support.

**Pull request closed without merging**